[WIP] [DO NOT MERGE] ML-RHDEVDOCS-5098:Remove all instances of .spec.dex parameter

masauso-lungu opened the pull request:
Jira issue : [RHDEVDOCS-5098](https://issues.redhat.com/browse/RHDEVDOCS-5098)

Aligned team: DevTools

OCP Version(s): 4.10 and later
**(Note: Changes to be applied in release GitOps v1.10)**

Docs preview: [Preview1](https://59959--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/configuring-sso-on-argo-cd-using-dex.html); [Preview2](https://59959--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/argo-cd-custom-resource-properties.html)

SME review: @svghadi 

QE review: @varshab1210

Peer review: @rolfedh

masauso-lungu commented:
Hello @svghadi ,

kindly review this PR for me.

rolfedh commented on modules/gitops-creating-a-new-client-in-dex.adoc:
```diff
@@ -4,27 +4,23 @@
+      openShiftOAuth: true <1>
+...
 ----
++
```
```suggestion
```
I think you can delete this line.

rolfedh commented on modules/gitops-disable-dex-using-spec-sso.adoc:
```diff
@@ -4,24 +4,6 @@
 :_content-type: PROCEDURE
 [id="gitops-disable-dex-using-spec-sso_{context}"]
-= Enabling or disabling Dex using .spec.sso
+= Disabling Dex using .spec.sso
```
Maybe 
```suggestion
= Disabling Dex by replacing .spec.sso
```

rolfedh reviewed (approved):
LGTM

kalexand-rh commented:
The `branch/enterprise-4.14` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.13. And any PR going into `main` must also target the latest version branch (enterprise-4.14).

If the update in your PR does NOT apply to version 4.14 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

svghadi commented:
Looks good so far. Thanks @masauso-lungu 

Just a few more changes
- We need to remove deprecated *\<Image\>*, *\<Resources\>*, *\<VerifyTLS\>* & *\<Version\>* fields from `SSO` row in [argo-cd-custom-resource-properties](https://docs.openshift.com/container-platform/4.13/cicd/gitops/argo-cd-custom-resource-properties.html#argo-cd-properties_argo-cd-custom-resource-properties)
- Fix a small typo in [configuring-sso-for-argo-cd-using-keycloak](https://docs.openshift.com/container-platform/4.13/cicd/gitops/configuring-sso-for-argo-cd-using-keycloak.html#gitops-creating-a-new-client-in-keycloak_configuring-sso-for-argo-cd-using-keycloak)
https://github.com/openshift/openshift-docs/blob/d4d32c0d1f7abca5d20252fa2958d577967ec8ef/modules/gitops-creating-a-new-client-using-keycloak.adoc?plain=1#L55
`.spec.keycloak.rootCA` should be `.spec.sso.keycloak.rootCA`

varshab1210 commented on cicd/gitops/configuring-sso-on-argo-cd-using-dex.adoc:
```diff
@@ -8,18 +8,13 @@ toc::[]
 
 After the {gitops-title} Operator is installed, Argo CD automatically creates a user with `admin` permissions. To manage multiple users, cluster administrators can use Argo CD to configure Single Sign-On (SSO).
 
-[IMPORTANT]
```
@svghadi As this is still supported until v1.10, should we be just changing the version here instead of removing this warning as a whole or it will be a release notes item?

svghadi replied:
Yes. Makes sense to keep it.

svghadi replied:
@masauso-lungu - Instead of completely removing `[IMPORTANT]` message, can we convert it to a `[note]` or `[warning]` having something like
> The `.spec.dex` parameter in the ArgoCD CR is no longer support from {gitops-title} v1.10 onward. Consider using the `.spec.sso` parameter instead.

Basically to inform users who are looking for older configurations.

svghadi commented on cicd/gitops/configuring-sso-on-argo-cd-using-dex.adoc:
```diff
@@ -8,9 +8,9 @@ toc::[]
 After the {gitops-title} Operator is installed, Argo CD automatically creates a user with `admin` permissions. To manage multiple users, cluster administrators can use Argo CD to configure Single Sign-On (SSO).
 
-[IMPORTANT]
+[WARNING]
```
Let's have this message as a [NOTE] instead of [WARNING]. Rest all looks good.

svghadi reviewed (approved):
Thank You @masauso-lungu

varshab1210 commented:
LGTM, thanks

Srivaralakshmi reviewed (approved):
@masauso-lungu Approved!

kalexand-rh commented:
Only versions 4.11+ are still in maintenance. I am removing labels from earlier versions from this PR.

masauso-lungu commented:
Please **DO NOT** merge this issue. The intention of this merge review is to keep the PRs merge ready with all approvals. The PR merging can happen only on the GA date (September 26)


/label merge-review-needed

mburke5678 commented on modules/gitops-creating-a-new-client-in-dex.adoc:
```diff
@@ -4,27 +4,22 @@
 = Enabling the Dex OpenShift OAuth Connector
 
-Dex uses the users and groups defined within OpenShift by checking the `OAuth` server provided by the platform. The following example shows the properties of Dex along with example configurations:
+Dex is installed by default for all the Argo CD instances created by the Operator. You can configure {gitops-title} to use Dex as the SSO authentication provider by setting the `.spec.sso` parameter. 
```
> created by the Operator

Do we need to state which Operator?

Srivaralakshmi replied:
@mburke5678 not required. It is understood as the GitOps Operator.

mburke5678 commented on modules/gitops-creating-a-new-client-in-dex.adoc:
```diff
@@ -4,27 +4,22 @@
-Dex uses the users and groups defined within OpenShift by checking the `OAuth` server provided by the platform. The following example shows the properties of Dex along with example configurations:
+Dex is installed by default for all the Argo CD instances created by the Operator. You can configure {gitops-title} to use Dex as the SSO authentication provider by setting the `.spec.sso` parameter. 
 
+Dex uses the users and groups defined within OpenShift by checking the `OAuth` server provided by the platform. 
```
s/OpenShift/{product-title}

mburke5678 commented on modules/gitops-creating-a-new-client-in-dex.adoc:
```diff
@@ -4,27 +4,22 @@
-  name: example-argocd
-  labels:
-    example: openshift-oauth
+...
```
s/.../# ...   
See the [doc guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#formatting-1).

mburke5678 commented on modules/gitops-creating-a-new-client-in-dex.adoc:
```diff
@@ -4,27 +4,22 @@
+    provider: dex
+    dex:
+      openShiftOAuth: true <1>
+...
```
s/.../# ...   
See the [doc guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#formatting-1).

mburke5678 commented on modules/gitops-disable-dex-using-spec-sso.adoc:
```diff
@@ -4,24 +4,6 @@
-+
-. To disable dex, either remove the `spec.sso` element from the Argo CD custom resource, or specify a different SSO provider.
\ No newline at end of file
+To disable dex, either remove the `spec.sso` element from the Argo CD custom resource, or specify a different SSO provider.
```
I don't think you need the comma in an either/or statement.
```suggestion
To disable dex, either remove the `spec.sso` element from the Argo CD custom resource or specify a different SSO provider.
```

mburke5678 commented:
A few nits. Otherwise merge review /lgtm

masauso-lungu commented:
Hello @mburke5678 , I have addressed your concerns, PTAL.

**Please DO NOT merge this issue. The intention for this merge review is to keep the PRs merge ready with all approvals. The PR merging can happen only on the GA date (September 27)**

/label merge-review-needed

Srivaralakshmi commented:
@masauso-lungu Please close this PR now.
